packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx

```diff
@@ -85,6 +87,32 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
     dateRange: searchParams.get('dateRange') as ThreadTimelineFilterTypes,
   });
 
+  const splitURLPath = useMemo(() => location.pathname.split('/'), []);
+  const decodedString = useMemo(
+    () => decodeURIComponent(splitURLPath[3]),
+    [splitURLPath],
+  );
+  const memoizedTopics = useMemo(() => topics, [topics]);
+  const testRoute = '/discussions';
+  console.log('splitURLPath: ', splitURLPath);
+  console.log('splitURLPath.length: ', splitURLPath.length);
+  //redirects users to All Discussions if they try to access a topic in the url that doesn't exist
+  useEffect(() => {
+    if (
+      decodedString &&
+      splitURLPath[2] === 'discussions' &&
+      splitURLPath.length === 4
```

Comment:
```
splitURLPath.length === 4
```

i think `splitURLPath` includes an empty string, thats why we check for `4`?

Reply:
Yes, when you go to any discussion topic that isn't All, Overview, or Archived there are 4